lib/synthesize-stream.ts

```diff
@@ -126,17 +126,42 @@ class SynthesizeStream extends Readable {
 
     socket.onmessage = message => {
       const chunk = message.data;
-      // some messages are strings - emit those unencoded, but push them to
-      // the stream as binary
-      const data = typeof chunk === 'string' ? chunk : Buffer.from(chunk);
+      // some info messages are sent as strings, telling the content_type and
+      // timings. Emit them as separate events, but do not send them along the
+      // pipe.
+      if (typeof chunk === 'string') {
+        try {
+          const json = JSON.parse(chunk);
+          if (json['binary_streams']) {
+            self.emit('binary_streams', message, chunk);
+          }
+          else if (json['marks']) {
+            self.emit('marks', message, chunk);
+          }
+          else if (json['words']) {
+            self.emit('words', message, chunk);
+          }
+          else if (json['error']) {
+            self.emit('error', message, chunk);
```

Comment:
This is the only one that makes me uneasy because the other existing error event has one arg, which is an Error object that's been formatted a little:
```ts
    socket.onerror = event => {
      const err = new Error('WebSocket connection error');
      err.name = SynthesizeStream.WEBSOCKET_CONNECTION_ERROR;
      err['event'] = event;
      self.emit('error', err);
      self.push(null);
    };
```

it is very likely that users already have handlers set up for the Error event expecting this type of object. I think it would be prudent here to either a) rename this event to something unique or b) try to put together a similarly formatted Error object to send back as the first arg. There are pros and cons to both - what do you think?

Reply:
I completely missed this, thanks for pointing it out. I think just joining them into one common event to listen for is probably best? The only thing missing for these types of errors would be a `name` to attribute to them, but it just be use add `WEBSOCKET_ERROR = 'Websocket error'` to the class definition and then rewrite this bit as:
```
json.name = SynthesizeStream.WEBSOCKET_ERROR;
json.event = message;
self.emit('error', err);
```

Reply:
Alright, so played around a bit more with this and it looks like for things like messing up the codec or forgetting `text` argument, the JSON object is sent over the `onmessage` channel to the user and then with an additional message on the `onclose` that says "see previous message for the error details". I'd probably lean towards making them emit on the same channel (and unify warnings to a similar emit format, though on a different channel), as the end result is all the same regardless of the exact reason.

Reply:
That sounds good to me, thanks for looking into it

Reply:
Everything else looks good. Once you update the error handler, I'll approve and merge 👍

Reply:
Alright, updated the error handler to look like the `onerror` signature. I did not touch the `warnings` one as I'm not sure there's a good way to mimic the warnings as JS does not really have a good concept of warnings like [python](https://docs.python.org/3/library/warnings.html#module-warnings) unfortunately.

Reply:
Yeah I think that's fine and there's no existing API there so I'm not worried about it